redfish-core/lib/log_services.hpp

```diff
@@ -1856,80 +2150,208 @@ inline void requestRoutesDBusEventLogEntry(App& app)
+                                       std::move(getHiddenProperty));
+            });
+}
 
+inline void requestRoutesDBusCELogEntry(App& app)
+{
+    BMCWEB_ROUTE(app, "/redfish/v1/Systems/system/LogServices/"
+                      "CELog/Entries/<str>/")
+        .privileges(redfish::privileges::getCELogEntry)
+        .methods(boost::beast::http::verb::get)(
+            [](const crow::Request&,
+               const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+               const std::string& param)
+
+            {
+                std::string entryID = param;
+                dbus::utility::escapePathForDbus(entryID);
+
+                // DBus implementation of EventLog/Entries
+                // Make call to Logging Service to find all log entry objects
                 crow::connections::systemBus->async_method_call(
-                    [asyncResp, entryId](const boost::system::error_code ec) {
+                    [asyncResp, entryID](const boost::system::error_code ec,
+                                         GetManagedPropertyType& resp) {
+                        if (ec.value() == EBADR)
+                        {
+                            messages::resourceNotFound(
+                                asyncResp->res, "EventLogEntry", entryID);
+                            return;
+                        }
                         if (ec)
                         {
-                            BMCWEB_LOG_DEBUG << "DBUS response error " << ec;
+                            BMCWEB_LOG_ERROR << "EventLogEntry (DBus) "
+                                                "resp_handler got error "
+                                             << ec;
                             messages::internalError(asyncResp->res);
                             return;
                         }
+                        getDBusLogEntry(asyncResp, resp, eventLogTypes::ceLog);
                     },
                     "xyz.openbmc_project.Logging",
-                    "/xyz/openbmc_project/logging/entry/" + entryId,
-                    "org.freedesktop.DBus.Properties", "Set",
-                    "xyz.openbmc_project.Logging.Entry", "Resolved",
-                    std::variant<bool>(*resolved));
+                    "/xyz/openbmc_project/logging/entry/" + entryID,
+                    "org.freedesktop.DBus.Properties", "GetAll", "");
             });
 
-    BMCWEB_ROUTE(
-        app, "/redfish/v1/Systems/system/LogServices/EventLog/Entries/<str>/")
-        .privileges(redfish::privileges::deleteLogEntry)
+    BMCWEB_ROUTE(app, "/redfish/v1/Systems/system/LogServices/"
+                      "CELog/Entries/<str>/")
+        .privileges(redfish::privileges::patchLogEntry)
+        .methods(boost::beast::http::verb::patch)(
+            [](const crow::Request& req,
+               const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+               const std::string& entryId) {
+                auto getHiddenProperty = [req, asyncResp,
```

Comment:
updateResolvedProperty() would be more accurate

Reply:
Does something like updateResolvedPropertyCallback work? I already have updateResolvedProperty() defined to do the actual update, and this method is common for both Event log and CE log.

Alternatively, I can make lambda as updateResolvedProperty and rename updateResolvedProperty() to something like doUpdateResolvedProperty()?

Reply:
Either work for me. 
updateResolvedPropertyCallback  although long seems fine.

Reply:
Done